app/helpers/rating_calculator.rb

```diff
@@ -0,0 +1,16 @@
+module RatingCalculator
+  def calc_add_to_average(restaurant, rating)
+    calc_average_after_update(restaurant,0, rating, (restaurant.num_of_reviews + 1))
+  end
+
+  def calc_sub_from_average(restaurant, rating)
+    num_of_reviews = (restaurant.num_of_reviews - 1) != 0 ? restaurant.num_of_reviews - 1 : 1
+    calc_average_after_update(restaurant, rating, 0, num_of_reviews)
+  end
+
+  def calc_average_after_update(restaurant, rating_was, rating, new_num_of_reviews = nil)
+    sum = restaurant.rating * restaurant.num_of_reviews - rating_was + rating
+
+    sum / new_num_of_reviews || restaurant.num_of_reviews
```

Comment:
You did allow a default `nil` value for `new_num_of_reviews `; In such case you would attempt to divide `sum / nil` which will throw an error.

Reply:
But there is an `||` operation, so it means if `new_num_of_reviews` is nil, it will take `restaurant.num_of_reviews`, won't it?